src/main/java/icoder/resources/TestResource.java

```diff
@@ -0,0 +1,142 @@
+    for (Folder mediaFolderByCountry : mediaFoldersByCountry) {
+      if (mediaFolderByCountry.getName().equalsIgnoreCase(folder.getName())) {
+        if (mediaFolderByCountry.getName().equals(folder.getName()))
+          continue;
+        else {
+          if (mediaFolderByCountry.getFullName() == null && folder.getFullName() == null) {
+            return true;
+          } else {
+            if (mediaFolderByCountry.getFullName() == null || folder.getFullName() == null) {
+              continue;
+            }
+            if (mediaFolderByCountry.getFullName().equals(folder.getFullName())) {
+              return true;
+            }
+          }
+        }
+      }
+    }
+    return false;
+  }
+
+    public Boolean checkIs1NameDuplicateInParent(Folder folder) {
+        List<Folder> mediaFoldersByCountry = new ArrayList<>();
+        for (Folder mediaFolderByCountry : mediaFoldersByCountry) {
+            if (mediaFolderByCountry.getName().equalsIgnoreCase(folder.getName())) {
+                if (mediaFolderByCountry.getName().equals(folder.getName()))
+                    continue;
+                else {
+                    if (mediaFolderByCountry.getFullName() == null && folder.getFullName() == null) {
+                        return true;
+                    } else {
+                        if (mediaFolderByCountry.getFullName() == null || folder.getFullName() == null) {
+                            continue;
+                        }
+                        if (mediaFolderByCountry.getFullName().equals(folder.getFullName())) {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
+        return false;
+    }
+    public Boolean checkIsNameDuplicateInPare3nt(Folder folder) {
+        List<Folder> mediaFoldersByCountry = new ArrayList<>();
+        for (Folder mediaFolderByCountry : mediaFoldersByCountry) {
+            if (mediaFolderByCountry.getName().equalsIgnoreCase(folder.getName())) {
+                if (mediaFolderByCountry.getName().equals(folder.getName()))
+                    continue;
+                else {
+                    if (mediaFolderByCountry.getFullName() == null && folder.getFullName() == null) {
+                        return true;
+                    } else {
+                        if (mediaFolderByCountry.getFullName() == null || folder.getFullName() == null) {
+                            continue;
+                        }
+                        if (mediaFolderByCountry.getFullName().equals(folder.getFullName())) {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
+        return false;
+    }
+    public Boolean checkIsNameDuplicateInParent2(Folder folder) {
+        List<Folder> mediaFoldersByCountry = new ArrayList<>();
+        for (Folder mediaFolderByCountry : mediaFoldersByCountry) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS135)